Index: materials/validators.py
===================================================================
diff --git a/materials/validators.py b/materials/validators.py
--- a/materials/validators.py	(revision ab75e64d16e8e70e144d8bcb613e386e0d7d93ee)
+++ b/materials/validators.py	(date 1774467397927)
@@ -1,0 +1,28 @@
+from urllib.parse import urlparse
+from rest_framework.serializers import ValidationError
+
+class YouTubeValidator:
+    def __init__(self, field):
+        self.field = field
+
+    def __call__(self, attrs):
+        link = attrs.get(self.field)
+
+        if not link:
+            return attrs
+
+        parsed_url = urlparse(link)
+        domain = parsed_url.netloc.lower()
+
+        allowed_domains = [
+            "youtube.com",
+            "www.youtube.com",
+            "youtu.be",
+        ]
+
+        if not any(domain == d or domain.endswith("." + d) for d in allowed_domains):
+            raise ValidationError(
+                {self.field: "Разрешены только ссылки на YouTube"}
+            )
+
+        return attrs
\ No newline at end of file
